src/app/admin/components/assign-role-modal/assign-role-modal.component.ts

```diff
@@ -0,0 +1,74 @@
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { map, Observable, startWith } from 'rxjs';
+import { Role, User } from '@app/auth/models';
+import { RoleModal } from '@app/admin/constants/role-modal.enum';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { AssignRoleModalData } from '@app/admin/types/assign-role-modal-data';
+import { ValidationService } from '@shared/services/validation/validation.service';
+import { UntilDestroy, untilDestroyed } from '@shared/until-destroy/until-destroy';
+
+@UntilDestroy
+@Component({
+  selector: 'lc-assign-role-modal',
+  templateUrl: './assign-role-modal.component.html',
+  styleUrls: ['./assign-role-modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AssignRoleModalComponent implements OnInit {
+  formName = 'role_assignment_form';
+  labels = RoleModal;
+  roles!: Role[];
+  form = this.formBuilder.group({
+    user: [null, Validators.required],
+    role: [null, Validators.required],
+  });
+  filteredOptions: Observable<User[]> | undefined;
```

Comment:
I think it shold be userControlValue$ according to the assigmemt operation

Reply:
named filteredUserOptions$ as in the assignment there is filtering